dspace-alb.yaml

```diff
@@ -1 +1,158 @@
-# TODO: draft this ALB template
\ No newline at end of file
+---
+  AWSTemplateFormatVersion: '2010-09-09'
+  Description: Template for creating our DSpace reverse proxy/load balancer
+
+  Parameters:
+    VpcId:
+      Type: AWS::EC2::VPC::Id
+      Description: The VPC ID where the load balancer will be created
+    SubnetIds:
+      Type: List<AWS::EC2::Subnet::Id>
+      Description: A list of subnet IDs where the load balancer will be created
+    LoadBalancerName:
+      Type: String
+      Description: A name for the load balancer
+    DSpaceEIP:
+      Type: String
+      Default: EIP allocation ID or IP address for this DSpace site
+    DomainName:
+      Type: String
+      Description: the domain name for this DSpace site
+    CertificateArn:
+      Type: String
+      Description: The ARN of the SSL/TLS certificate to use for HTTPS listener
+    ApiInstancePort:
+      Type: Number
+      Description: The port number to forward /server/* and /server URL patterns to on the API instance
+    ClientInstancePort:
+      Type: Number
+      Description: The port number to forward all other URL patterns to on the client instance
+    InstanceSecurityGroup:
+      Type: AWS::EC2::SecurityGroup::Id
+      Description: The ID of the security group to apply to both the API and client instances
+    LoadBalancerSecurityGroup:
+      Type: AWS::EC2::SecurityGroup::Id
+      Description: The ID of the security group to apply to the load balancer
+
+  Metadata:
+    UnusedParams: # cfn-lint ignore
+      ApiInstancePort: !Ref ApiInstancePort
+      ClientInstancePort: !Ref ClientInstancePort
+      InstanceSecurityGroup: !Ref InstanceSecurityGroup
+      LoadBalancerSecurityGroup: !Ref LoadBalancerSecurityGroup
+      DSpaceEIP: !Ref DSpaceEIP
+
+  Resources:
+    # 1. Create the Elastic Load Balancer resource
+    LoadBalancer:
+      Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+      Properties:
+        Name: !Ref LoadBalancerName
```

Comment:
Is there a specific reason to have the name passed in at runtime? I feel like we'd like to keep a consistent naming pattern across dev / stage / prod which is harder to enforce if we pass it in here, right?

Reply:
it's a placeholder, I intend to build a name out of the other parameters